interfacer/src/browsh/tty.go

```diff
@@ -139,34 +147,66 @@ func isMultiLineEnter(ev *tcell.EventKey) bool {
 	return activeInputBox.isMultiLine() && ev.Key() == 13 && ev.Modifiers() != 4
 }
 
-func handleScrolling(ev *tcell.EventKey) {
+func generateLeftClickYHack(x, y int, yHack bool) {
```

Comment:
Can you just remind me of this hack again? Is it to do with doubling the y-coord? We should either fix this properly now, or have a good reason not to fix it.

Reply:
This hack isn't strictly needed at the moment, however I think that a need for something like it will arise sooner or later. Basically the issue is that browsh partitions the screen into a number of columns and rows (grid) and that these rows times columns areas are the only places that can be clicked with the mouse per default. Sometimes user elements (like an input field, clickable image or the like) in the browser backend aren't placed exactly in the middle of a row/cell pair, but instead in the middle of two neighbouring cells. This hack allows to generate a click in between them and therefore increases the click resolution so to speak, so maybe this shouldn't even be called hack and another name instead. But looking at the code it definitely looks hackish, so the name came to me easily. The reason I created this was that I had input fields that weren't properly aligned in the grid and therefore it wasn't straightforward to translate clicks from the link hint data (generated by the backend and sent to the tty for processing) that has a higher resolution (pixels) so to speak into clicks on the tty (with the lower tty resolution). I think this will prove useful again, even though it looks unpleasant.

Reply:
Ah right I understand now, so the hack just allows you to click _in the middle_ of a TTY cell, rather than at the top right? So is there a reason to not actually just make this the default? Considering the code I've already written I think defaulting to clicking in the middle will be fine.

Reply:
@tombh 
It allows to generate a click, that's between two cells (the cell the click is generated for and the cell directly below(y) it), and that's what I referred to as the middle. From the top of my head I can't tell you why this shouldn't be the default, but I feel it'd be good to have more control over where generated clicks in the tty go when they're received in the backend. The terminal mouse has a low resolution anyway, so picking the middle of the cell seems reasonable for this translation.
Defaulting to the middle might indeed be fine, but I needed this YHack only to fix a corner case, so hopefully we'll not run into new corner cases where we need the old version of the code again.

Reply:
If you remember what the corner case is then we can write a test and then hopefully any problems arising for switching the default to this new behaviour will show up.

Reply:
Oh and do you remember why 2 mouse events with a pause in between are sent for each click?

Reply:
@tombh 
One is for pressing the button, and one is for releasing it. Generating a single click. The pause is there to make it appear human-like. I tested different setups, and this one seemed to work by far the best, while it allowed me to reuse existing code.

Reply:
Ah yes of course!